app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,55 @@
+# user_reviews GET    /users/:user_id/reviews(.:format)                reviews#index
+#              POST   /users/:user_id/reviews(.:format)                reviews#create
+# new_user_review GET    /users/:user_id/reviews/new(.:format)            reviews#new
+# edit_user_review GET    /users/:user_id/reviews/:id/edit(.:format)       reviews#edit
+#  user_review GET    /users/:user_id/reviews/:id(.:format)            reviews#show
+#              PATCH  /users/:user_id/reviews/:id(.:format)            reviews#update
+#              PUT    /users/:user_id/reviews/:id(.:format)            reviews#update
+#              DELETE /users/:user_id/reviews/:id(.:format)            reviews#destroy
+#
+#
+#              product_reviews GET    /products/:product_id/reviews(.:format)          reviews#index
+#                 POST   /products/:product_id/reviews(.:format)          reviews#create
+# new_product_review GET    /products/:product_id/reviews/new(.:format)      reviews#new
+# edit_product_review GET    /products/:product_id/reviews/:id/edit(.:format) reviews#edit
+#  product_review GET    /products/:product_id/reviews/:id(.:format)      reviews#show
+#                 PATCH  /products/:product_id/reviews/:id(.:format)      reviews#update
+#                 PUT    /products/:product_id/reviews/:id(.:format)      reviews#update
+#                 DELETE /products/:product_id/reviews/:id(.:format)      reviews#destroy
+class ReviewsController < ApplicationController
+
+  helper_method :top_ratings, :display_name
+
+  def new
+    @review = Review.new
+    @product = Product.find(params[:product_id])
+  end
+
+  def create
+    @product = Product.find(params[:review][:product_id])
+    @review = Review.create(review_edit_params[:review])
+    if @review.save
+      redirect_to product_path(@review.product_id)
+    else
+      flash.now[:alert] = 'Review could not be saved.'
```

Comment:
It is often good to provide the user with recommendations to ensure they can be successful the next time around. The potential problems that users could solve are usually found by looking at the validations.